recipe/session/recipeImplementation.go

```diff
@@ -39,27 +39,102 @@ var protectedProps = []string{
 	"antiCsrfToken",
 }
 
-var JWKCacheMaxAgeInMs = 60000
+var JWKCacheMaxAgeInMs int64 = 60000
 var JWKRefreshRateLimit = 500
-var jwksResults []sessmodels.GetJWKSResult
 
-func getJWKS() []sessmodels.GetJWKSResult {
-	result := []sessmodels.GetJWKSResult{}
+// Maintains a map of the core path to the result
+var jwksCache map[string]sessmodels.GetJWKSResult = map[string]sessmodels.GetJWKSResult{}
+
+func getJWKS() []sessmodels.GetJWKSFunctionObject {
```

Comment:
make this return a result directly as opposed to a function that returns a result